src/braket/pennylane_plugin/translation.py

```diff
@@ -576,6 +576,8 @@ def translate_result_type(  # noqa: C901
     observable = measurement.obs
 
     if return_type is ObservableReturnTypes.Probability:
+        if observable:
+            raise qml.DeviceError("Probability result type not supported for observables")
```

Comment:
This looks to be causing some issues for the tests. Is this the expected outcome now?

Reply:
So at the moment the pl-device-tests are xfailing the hadamard gradient tests due to the incorrect results.  We may instead need to xfail at a different location as now we get an error on execution, instead of on result comparison:

https://github.com/PennyLaneAI/pennylane/blob/883c3ed0a66d06841e074c90f59317972af36cee/pennylane/devices/tests/test_gradients_autograd.py#L141